src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -65,22 +66,32 @@ public async Task<IActionResult> Get(string name)
         [HttpPut]
         [Route("admin/functions/{name}")]
         [Authorize(Policy = PolicyNames.AdminAuthLevel)]
-        public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] FunctionMetadataResponse functionMetadata)
+        public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] FunctionMetadataResponse functionMetadata, [FromServices] IEnumerable<IHostedService> hostedServices)
```

Comment:
Instead of taking the entire collection of hosted services, let's introduce an interface for the file monitoring service and take a dependency on that instead.

Reply:
we still need `FileMonitoringService` added as `IHostedService` to be a background task. I tried couple of ways:

1.  
```
                    services.AddSingleton<FileMonitoringService>();
                    services.AddSingleton<IHostedService, FileMonitoringService>(serviceProvider =>
                    {
                        return serviceProvider.GetService<FileMonitoringService>();
                    });
```
in this case we need to remove the line from DependencyValidator as FileMonitoringService as IHostedService will be resolved letter.
https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs#L48
2. Add a helper class - BackgroundServiceStarter:
https://stackoverflow.com/questions/51254053/how-to-inject-a-reference-to-a-specific-ihostedservice-implementation

So I decided to extract the service from` IEnumerable<IHostedService>`. Please advise which direction should we go.

Reply:
Oh I see the comment (GitHub put it at the bottom)

In addition to the resolution as a hosted service, you can add it as an `IFileMonitoringService` as well.